Nodes Recovery: add infrastructure info to DB

jrochas opened the pull request:
- add column holding infrastructure runtime variable map
- refactor all infrastructures such that their runtime variables are now stored and updated in the map. Infrastructures use now read/write locks to synchronize variable accesses.
- persist the update of runtime variables for all infrastructures
- add and align unit tests

jrochas committed:
Nodes Recovery: add infrastructure info to DB

- add column holding infrastructure runtime variable map
- refactor all infrastructures such that their runtime variables are now stored and updated in the map. Infrastructures use now read/write locks to synchronize variable accesses.
- persist the update of runtime variables for all infrastructures
- add and align unit tests

marcocast commented on rm/rm-server/src/test/java/org/ow2/proactive/resourcemanager/db/NodeSourcesTest.java:
```diff
@@ -43,13 +46,22 @@
+        infrastructureVariables.put(INFRASTRUCTURE_VARIABLE_KEY, INFRASTRUCTURE_VARIABLE_VALUE);
     }
 
     @After
```
Is this @after necessary?

jrochas replied:
I think it is useful in order to release the dbManager object properly after each test.

jrochas committed:
Nodes Recovery: fix issues for infrastructure variables initialization

- fix call to super.initialize for SSH infrastructures
- fix persistence of infrastructure variable for the first time when the node source of the infrastructure is set
- when node sources are removed, now their nodes are removed first.
- fix race condition between the execution flow that shuts down the node source and the execution flow that removes the node source from the database.

marcocast commented on rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/db/NodeSourceData.java:
```diff
@@ -61,6 +62,11 @@
+    /**
+     * name of the variable --> value of the variable
+     */
+    private Map<String, Object> infrastructureVariables;
```
maybe it's better to initialise this with an empty map to avoid possible nullpointer exceptions

marcocast commented on rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/HostsFileBasedInfrastructureManager.java:
```diff
@@ -373,6 +366,56 @@ private void removeNodes(List<String> depNodeURLs) {
+        runtimeVariables.put(PN_TIMEOUT_KEY, new HashMap<String, Boolean>());
+    }
+
+    private void atomicRemoveNode(final Node node) {
```
This method is a bit too long, is it possible to refactor it into multiple short private methods with descriptive names?

marcocast reviewed (approved):
just few refactoring but LGTM
